src/app-layer-ftp.c

```diff
@@ -397,13 +400,18 @@ static int FTPGetLineForDirection(FtpState *state, FtpLineState *line_state)
          * if we see fragmentation then it's definitely something you
          * should alert about */
         if (line_state->current_line_db == 0) {
-            line_state->db = FTPMalloc(state->input_len);
+            int32_t input_len = state->input_len;
```

Comment:
Why is `input_len` signed ?

Reply:
Only because its source, `state->input_len` is in32_t.

Reply:
Indeed, but why is `state->input_len` signed in the first place ?

Reply:
> Indeed, but why is `state->input_len` signed in the first place ?

```
 ¯\_(ツ)_/¯
```

Reply:
> So, I take this is not the good opportunity fo fix this... Should I create a ticket for it for later ?

Yeah, I'd rather keep the fix as focused as possible for now.

Reply:
https://redmine.openinfosecfoundation.org/issues/5311